### PR TITLE
Migrate site to Astro

### PR DESCRIPTION
**Pull request by lucperkins**
The [iles](https://iles.pages.dev/) framework has essentially been abandoned so it's time to re-platform.

TODOs:

- [x] Style the language selector components
- [x] Adjust font size in code samples
- [x] Markdown plugin for auto-rendering emoji (like `:rocket:`)
- [x] Fix font spacing in navbar links
- [x] Add titles to all code snippets
- [x] Make sure Posthog is working as expected
- [x] Fix external sources components on Learn More page
- [x] Styling for Nix store path component
- [x] Fix line thickness in admonition icons (and elsewhere if possible)
- [x] Link icons in "Related" and "External sources" section of concept pages
- [x] Cookie consent
- [x] Drawer toggle

### CODE REVIEW

**Comment by grahamc** [comment]
holy guacamole

**Comment by grahamc** [comment]
Off the bat, WOW!

I noticed the rounded pill-buttons change color on hover before they're actually clickable:

CleanShot 2024-11-19 at 17 20 14@2x

**Comment by lucperkins** [comment]
@grahamc Fixed

File: src/components/CookieConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { posthog } from "posthog-js";
+import { useEffect, useState } from "react";
+
+const CookieConsent = () => {
+  const [show, setShow] = useState<boolean>(false);
+  const [optionSelected, setOptionSelected] = useState<boolean>(false);
+
+  useEffect(() => {
+    posthog.init("phc_OPJtdGL4gAGdo8VKLsHz4LmKfoOMKkrza1BsBNeUdx4", {
```
Comment (grahamc) [inline src/components/CookieConsent.tsx:9]:
```suggestion
    posthog.init("phc_qZk1aHPpXUq0ohRcTjFL5w8Tpo5Asht9UhFBjzruBFD", {
```

Might as well update the project while we're here.

**Comment by grahamc** [comment]
The feedback widget needs a bit of touch-ups:

old:

CleanShot 2024-11-20 at 08 56 09@2x


new:

CleanShot 2024-11-20 at 08 55 35@2x



Note this is a problem on both sites, and I'm not real fussed, but I noticed it expands when typing:

CleanShot 2024-11-20 at 08 57 01

...and with a narrower viewport, it gets a bit weirder:

CleanShot 2024-11-20 at 08 59 20@2x

File: src/components/CookieConsent.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { posthog } from "posthog-js";
+import { useEffect, useState } from "react";
+
+const CookieConsent = () => {
```
Comment (grahamc) [inline src/components/CookieConsent.tsx:4]: I'm not seeing the cookie consent banner show up, but I am seeing event data going to PostHog.

File: src/components/CookieConsent.tsx (hunk 1 above)
Comment (lucperkins) [reply src/components/CookieConsent.tsx:9]: Update Posthog project key

**Comment by grahamc** [comment]
This is looking good to me. I'm still not seeing the cookie consent popup, but beyond that, I don't see any obvious problems. I think we should move forward with a merge as soon as the consent window comes up.

**Comment by grahamc** [comment]
Have you poked around this repo? https://github.com/jop-software/astro-cookieconsent/blob/main/src/main.ts


...poking around a bit I noticed a weird flash of content, which turns out to be all the user-initiated things:

CleanShot 2024-11-22 at 00 06 21@2x

**Comment by lucperkins** [comment]
@grahamc There isn't really a way out of the flash of unstyled content with dark mode and a fully static site. Vue has some JS magic built into it that prevents that but we don't have access to that here.

**Review by flexiondotorg — APPROVED** [review]
I completed a side-by-side read-through, and the Astro version is spot on.

**Comment by grahamc** [comment]
@lucperkins I think if you pin posthog to just before [v1.136.0](https://github.com/PostHog/posthog-js/releases/tag/v1.136.0) our posthog conditions will work correctly.

**Comment by lucperkins** [comment]
@grahamc Alas, that doesn't work either. I just cannot get `posthog.has_opted_in_capturing() || posthog.has_opted_out_capturing()` to be anything but `true` no matter what I do. Very frustrating. Gonna keep trying.

**Comment by grahamc** [comment]
@lucperkins I think it is because posthog is getting its .js from us.i.posthog.com, instead of the bundled version we've pinned. Is that possible?

**Pull request merged**